src/main/java/com/puppycrawl/tools/checkstyle/xpath/AttributeNode.java

```diff
@@ -0,0 +1,156 @@
+////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2017 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.xpath;
+
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import net.sf.saxon.om.NodeInfo;
+import net.sf.saxon.tree.iter.AxisIterator;
+import net.sf.saxon.type.Type;
+
+/**
+ * Represents attribute of the element.
+ *
+ * @author Timur Tibeyev
+ */
+public class AttributeNode extends AbstractNode {
+    /** The name of the attribute. */
+    private final String name;
+
+    /** The value of the attribute. */
+    private final String value;
+
+    /**
+     * Creates a new {@code AttributeNode} instance.
+     *
+     * @param name name of the attribute
+     * @param value value of the attribute
+     */
+    public AttributeNode(String name, String value) {
+        this.name = name;
+        this.value = value;
+    }
+
+    /**
+     * Returns attribute value.
+     * @param namespace namespace
+     * @param localPart actual name of the attribute
+     * @return attribute value
+     */
+    @Override
+    public String getAttributeValue(String namespace, String localPart) {
+        throw throwUnsupportedOperationException();
+    }
```

Comment:
I still see a lot of `throwUnsupportedOperationException` in `AttributeNode`.
If that functionality is not required for all nodes - please explain this in javadoc. If smb is overriding it, it can not stay in abstract class, and it is better to keep such methods in exact classes that are implementable.